ai-ml/ray/terraform/main.tf

```diff
@@ -0,0 +1,408 @@
+  region = "us-east-1"
+  alias  = "virginia"
+}
+
+provider "kubernetes" {
+  host                   = module.eks.cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", module.eks.cluster_name]
+  }
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = module.eks.cluster_endpoint
+    cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+
+    exec {
+      api_version = "client.authentication.k8s.io/v1beta1"
+      command     = "aws"
+      # This requires the awscli to be installed locally where Terraform is executed
+      args = ["eks", "get-token", "--cluster-name", module.eks.cluster_name]
+    }
+  }
+}
+
+provider "kubectl" {
+  apply_retry_count      = 5
+  host                   = module.eks.cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+  load_config_file       = false
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", module.eks.cluster_name]
+  }
+}
+
+#---------------------------------------------------------------
+# Data Sources
+#---------------------------------------------------------------
+
+data "aws_availability_zones" "available" {}
+
+# Used for Karpenter Helm chart
+data "aws_ecrpublic_authorization_token" "token" {
+  provider = aws.virginia
+}
+
+#---------------------------------------------------------------
+# Locals
+#---------------------------------------------------------------
+
+locals {
+  name   = var.name
+  region = var.region
+
+  vpc_cidr           = "10.0.0.0/16"
+  secondary_vpc_cidr = "100.64.0.0/16"
+  azs                = slice(data.aws_availability_zones.available.names, 0, 3)
+
+  cluster_version = var.eks_cluster_version
+
+  tags = {
+    Blueprint  = local.name
+    GithubRepo = "github.com/awslabs/data-on-eks"
+  }
+}
+
+#---------------------------------------------------------------
+# VPC
+#---------------------------------------------------------------
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 3.0"
+
+  name = local.name
+  cidr = local.vpc_cidr
+
+  secondary_cidr_blocks = [local.secondary_vpc_cidr]
+
+  azs = local.azs
+  private_subnets = concat(
+    [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 4, k)],
+    [for k, v in local.azs : cidrsubnet(local.secondary_vpc_cidr, 4, k)]
+  )
+  public_subnets = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k + 48)]
+  # Control Plane Subnets
+  intra_subnets = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k + 52)]
```

Comment:
Curious about `intra_subnets`. Any documentation around this and why is it recommended? Or Add some comments just above that.